geowebcache/core/src/main/java/org/geowebcache/seed/TruncateTask.java

```diff
@@ -49,11 +56,40 @@ public TruncateTask(StorageBroker sb, TileRange tr, TileLayer tl, boolean doFilt
     protected void doActionInternal() throws GeoWebCacheException, InterruptedException {
         super.state = GWCTask.STATE.RUNNING;
         checkInterrupted();
-        try {
-            storageBroker.delete(tr);
-        } catch (Exception e) {
-            super.state = GWCTask.STATE.DEAD;
-            log.error("During truncate request", e);
+
+        final String layerName = tl.getName();
+        TileRange tr = trIter.getTileRange();
+        checkInterrupted();
+
+        long[] gridLoc = trIter.nextMetaGridLocation(new long[3]);
+
+        while (gridLoc != null && !this.terminate) {
```

Comment:
Won't this make the code iterate over tiles that do not exist? What if one truncates a EPSG:900913 with 30 zoom levels,
won't this take ages to finish? The current code won't even visit a zoom level directory if it's not there.